### PR TITLE
Add several new R packages

### PR DESCRIPTION
**Pull request by adamjstewart**
Also updates to the latest version for several existing R packages.

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-rminer/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    depends_on('r-mda', type=('build', 'run'))
+    depends_on('r-rpart', type=('build', 'run'))
+    depends_on('r-randomforest', type=('build', 'run'))
+    depends_on('r-adabag', type=('build', 'run'))
+    depends_on('r-party', type=('build', 'run'))
+    depends_on('r-cubist', type=('build', 'run'))
+    depends_on('r-kernlab', type=('build', 'run'))
+    depends_on('r-e1071', type=('build', 'run'))
+    depends_on('r-glmnet', type=('build', 'run'))
+    depends_on('r-xgboost', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-rminer/package.py]: Concretizing this package is absolutely brutal:
```
$ time spack spec r-rminer
...
real	7m16.433s
user	7m9.730s
sys	0m4.468s
```

**Comment by citibeth** [comment]
Why do the new packages subclass `Package` and not `RPackage`?

**Comment by adamjstewart** [comment]
Hmm, that is a good question, seeing as how it correctly renamed them to `r-*`. I'll take a look and fix that up. Good catch!

File: lib/spack/spack/cmd/create.py [hunk 2]
```diff
@@ -249,6 +249,7 @@ def __init__(self, name, *args):
 
 class RPackageTemplate(PackageTemplate):
     """Provides appropriate overrides for R extensions"""
+    base_class_name = 'RPackage'
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/create.py]: Somehow I completely forgot this in #2761.

**Comment by tgamblin** [comment]
@adamjstewart: is this WIP?

**Comment by adamjstewart** [comment]
Well, I haven't yet tried to install any of them, but they shouldn't break anything at least. I was going to install them on Monday 😄

**Comment by tgamblin** [comment]
Ok I'll wait til Monday.

**Comment by adamjstewart** [comment]
Ok, this should be all of the packages I need for now. I was able to install all of them, but I wasn't able to activate them due to the bug I mentioned in https://github.com/LLNL/spack/issues/2951#issuecomment-276210589.

**Pull request merged**